src/DotNetWorker/Converters/JsonPocoConverter.cs

```diff
@@ -1,19 +1,23 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Text.Json;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Azure.Core.Serialization;
+using Microsoft.Azure.Functions.Worker.Configuration;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.Functions.Worker.Converters
 {
     internal class JsonPocoConverter : IConverter
     {
-        private readonly IOptions<JsonSerializerOptions> _options;
+        private readonly ObjectSerializer _serializer;
 
-        public JsonPocoConverter(IOptions<JsonSerializerOptions> options)
+        public JsonPocoConverter(IOptions<WorkerOptions> options)
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _serializer = options?.Value?.Serializer ?? throw new ArgumentNullException(nameof(options));
```

Comment:
The exception only applies if the parameter is null, for the properties, we should throw a more descriptive exception.